### ISSUE TITLE
Add hint when trying to import a data constructor as if it is a type

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript#3925]
## Summary

Add a hint to make it clearer what has gone wrong and how to fix it if the user tries to import a data constructor as if it were a type.

## Motivation

The distinction between data types (which live at the type level) and data constructors (which live at the term/value level) isn't always clear to people. One way this manifests is by people trying to import constructors as if they are types. For example, someone might write:

```
import Data.Maybe (Just)
```

which produces an error along the lines of:

> Cannot import type Just from module Data.Maybe

which isn't always easy for people to make sense of.

## Proposal

If a user tries to import a type which doesn't exist, but a data constructor of the given name does exist in that module, add a hint to the error along the lines of:

> Just is a data constructor of the type Maybe. To import the data constructor Just, use:
>
>     import Data.Maybe (Maybe(..))

**Comment by afcondon** [comment]
Maybe worth really spelling it out that the `(..)` is syntax for "all the constructors"? 

Like, maybe, "To import Just, as well as the other constructors of type Maybe, use:"

**Comment by hdgarrood** [comment]
Yes, that’s true. I think 99% of the time you do want to import all of the constructors, which is why I suggested using Maybe(..) rather than Maybe(Just), but we should say that it will import all of the constructors with the (..) syntax.
